src/ios-scroll.js

```diff
@@ -0,0 +1,557 @@
+    const fixedOffset = y === 0 ? 0 : y + this.scrollTop_;
+    const fixeds = this.fixedElementStyles_;
+    for (let i = 0; i < fixeds.length; i++) {
+      fixeds[i].transform = `translate3d(0, ${fixedOffset}px, 0)`;
+    }
+
+    if (scrollEventDelta > SCROLL_EVENT_DELTA) {
+      this.lastScrollEventOffset_ = y;
+      // The only time we scroll to 0 is when we're synchronizing the scrollTop
+      // and synthetic scroll. In that case, the body's own scroll event will
+      // fire.
+      if (y !== 0) {
+        this.scrollObservable_.fire();
+      }
+    }
+
+    return true;
+  }
+
+  /**
+   * Determines the y position of a touch event relative to the screen.
+   *
+   * @param {!TouchEvent} event
+   * @return {number}
+   */
+  eventY_(event) {
+    // Surprisingly, `touchend` does not provide a Touch inside `touches`,
+    // but every `TouchEvent` will provide one inside `changedTouches`.
+    return event.changedTouches[0].screenY;
+  }
+
+  /**
+   * Tracks the velocity of touch movement using a sliding equation.
+   *
+   * @param {number} delta the change in y position since the last velocity
+   *     tracking.
+   */
+  track_(delta) {
+    const now = Date.now();
+    const elapsed = now - this.timestamp_;
+    this.timestamp_ = now;
+
+    // An arbitrary sliding acceleration equation to gain speed based on the
+    // last velocity.
+    // Constants have been pulled out to highlight that they may change.
+    const pixelsPerDelta = 500;
+    const currentWeight = 0.8;
+    const lastWeight = 1 - currentWeight;
+    const v = pixelsPerDelta * delta / (1 + elapsed);
+    this.velocity_ = (currentWeight * v) + (lastWeight * this.velocity_);
+  }
+
+  /**
+   * A step function that simulates momentum using a velocity and total offset
+   * amplitude. We approach the target offset exponentially, backing off as we
+   * get closer.
+   */
+  autoScroll_() {
+    this.autoScrolling_ = 0;
+
+    // We approach the target exponentially, slowing down as we get closer.
+    // Note that duration is arbitrary, it just "feels" right.
+    const duration = 450;
+    const elapsed = Date.now() - this.timestamp_;
+    const delta = this.amplitude_ * Math.exp(-elapsed / duration);
+
+    if (Math.abs(delta) > 0.3) {
+      if (this.scrollTo_(this.target_ - delta)) {
+        this.autoScrolling_ = requestAnimationFrame(this.boundAutoScroll);
+        return;
+      }
+    } else {
+      this.scrollTo_(this.target_);
+    }
+    this.scrollStopped_();
+  }
+
+  /**
+   * Called when the doc first starts scrolling (ie, there is no scroll
+   * synthetic scroll momentum). This will be called if the user natively
+   * scrolls the document, then continues the scroll outside of the iframe.
+   *
+   * @param {TouchEvent} event
+   */
+  scrollStarted_(event) {
+    if (this.bodyScrolled_) {
+      // If the user is still scrolling natively, this will stop it and start
+      // the synthetic scrolling.
+      event.preventDefault();
```

Comment:
why not do something like - overflow:hidden to stop scrolling? - i dont see anything wrong here. just asking